lib/ex_unit/lib/ex_unit/callbacks.ex

```diff
@@ -397,20 +395,7 @@ defmodule ExUnit.Callbacks do
   @spec start_supervised(Supervisor.child_spec() | module | {module, term}, keyword) ::
           Supervisor.on_start_child()
   def start_supervised(child_spec_or_module, opts \\ []) do
-    sup =
-      case ExUnit.OnExitHandler.get_supervisor(self()) do
-        {:ok, nil} ->
-          {:ok, sup} = Supervisor.start_link([], @supervisor_opts)
-          ExUnit.OnExitHandler.put_supervisor(self(), sup)
-          sup
-
-        {:ok, sup} ->
-          sup
-
-        :error ->
-          raise ArgumentError, "start_supervised/2 can only be invoked from the test process"
```

Comment:
see the test failure: https://github.com/elixir-lang/elixir/pull/10035/checks?check_run_id=675089179 (I will fix it soon), I was worried that by changing the exception message this may become confusing. wdyt?

Reply:
We can refer to the functions that may call it. Something like "fetch_test_supervisor!/start_supervised/start_supervised! can only ....".

Reply:
@wojtekmach another option is to name it `fetch_test_supervisor`, return `{:ok, ...} | :error` and always let the caller raise. This may be cleaner after all. Thoughts?